src/Raven.Server/ServerWide/Commands/UpdatePullReplicationAsHubCommand.cs

```diff
@@ -22,7 +23,13 @@ public override string UpdateDatabaseRecord(DatabaseRecord record, long etag)
             }
             else
             {
-                PullReplicationDefinition.RemoveHub(record.HubPullReplications, Definition.TaskId);
+                foreach (var task in record.HubPullReplications)
+                {
+                    if (task.TaskId != Definition.TaskId)
+                        continue;
+                    record.HubPullReplications.Remove(task);
```

Comment:
not sure if it is okay to remove from list while iterating.

Reply:
See the next line, we `break` immediately afterward, no issue